File: src/radarr.go
package main

type RadarrMovie struct {
	Title      string `json:"title" binding:"required"`
	FolderPath string `json:"folderPath" binding:"required"`
}

type RadarrMovieFile struct {
	RelativePath string `json:"relativePath" binding:"required"`
}

type RadarrEvent struct {
	Movie     *RadarrMovie     `json:"movie" binding:"required"`
	MovieFile *RadarrMovieFile `json:"movieFile" binding:"required"`
}

File: api/src/models/radarr.rs
use serde::{Deserialize, Serialize};

#[derive(Serialize, Deserialize)]
pub struct RadarrMovie {
    pub title: String,

    pub year: u32,

    #[serde(rename = "folderPath")]
    pub folder_path: String,
}

#[derive(Serialize, Deserialize)]
pub struct RadarrMovieFile {
    #[serde(rename = "relativePath")]
    pub relative_path: String,
}

#[derive(Serialize, Deserialize)]
pub struct RadarrWebhook {
    #[serde(rename = "eventType")]
    pub event_type: String,

    pub movie: RadarrMovie,

    #[serde(rename = "movieFile")]
    pub movie_file: Option<RadarrMovieFile>,
}
